src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1287,7 +1287,9 @@ private ISpreadOperation CreateBoundCollectionExpressionSpreadElement(BoundColle
             SyntaxNode syntax = element.Syntax;
             bool isImplicit = element.WasCompilerGenerated;
             var elementType = element.EnumeratorInfoOpt?.ElementType.GetPublicSymbol();
-            var elementConversion = BoundNode.GetConversion(iteratorItem, element.ElementPlaceholder);
+            var elementConversion = iteratorItem?.HasErrors == true ?
```

Comment:
This doesn't feel correct. Whether or not the iterator item has errors seems like it should be irrelevant to whether a conversion is present.

Reply:
I also don't see any tests related to this change?

Reply:
The issue is hit in one of the new tests in the IOperation test run, but I've added a specific test: `IOperation_SpreadElementConversion_Untyped`.

Reply:
> This doesn't feel correct. Whether or not the iterator item has errors seems like it should be irrelevant to whether a conversion is present.

Good catch, thanks. I've updated the check to look for collection initializers specifically.

Reply:
I've split out the IOperation fix to https://github.com/dotnet/roslyn/pull/71461. This PR is now limited to tests for https://github.com/dotnet/roslyn/issues/69704.